pkg/osquery/runtime/runtime.go

```diff
@@ -587,6 +587,17 @@ func (r *Runner) launchOsqueryInstance() error {
 		autoupdate.DeleteOldUpdates(),
 	)
 
+	// The extensions file should be owned by the process's UID or by root.
+	// Osquery will refuse to load the extension otherwise. Also ensure osqueryd
+	// is owned by root or current uid.
+	err = ensureProperPermissions(o, []string{paths.extensionPath, currentOsquerydBinaryPath})
```

Comment:
instead of the current osqueryd binary path, we could export `autoupdate.getUpdateDir`, and pass the result of that into this ensureProperPermissions func. @directionless any objections to that?